packages/jupyter-ai/jupyter_ai/handlers.py

```diff
@@ -217,24 +231,39 @@ async def on_message(self, message):
             self.log.error(e)
             return
 
+        # message sent to the agent instance
         message = HumanMessage(
             content=chat_request.prompt,
             additional_kwargs=dict(user=asdict(self.current_user))
```

Comment:
This might not be required anymore, since the chat history for the extension is maintained outside of the memory now. Consequently, you might not even need creating a HumanMessage here, can just pass the prompt directly to the chat provider.

Reply:
Probably. The HumanMessage object doesn't appear to be all that useful; it's essentially a prompt template, right?

Reply:
Yes, this is used mostly by the chat message history to serialize/deserialize user prompts. Not much utility here, since we added our own chat message objects.